state-synchronizer/src/executor_proxy.rs

```diff
@@ -96,21 +91,21 @@ impl ExecutorProxy {
 
     // TODO make this into more general trait method in `on_chain_config.rs`
     // once `StorageRead` trait is replaced with `LibraDBTrait` and `batch_fetch_config` method is no longer async
-    async fn fetch_all_configs(&self) -> HashMap<ConfigID, Option<Vec<u8>>> {
+    async fn fetch_all_configs(&self) -> Result<HashMap<ConfigID, Vec<u8>>> {
```

Comment:
I'm a bit concerned about the use of HashMap here due to its non-deterministic nature. I guess it's fine in this case as each different component should be independent of each other theoretically so iteration order shouldn't matter that much.